backend/fms_core/viewsets/project.py

```diff
@@ -59,4 +62,26 @@ def summary(self, _request):
             "total_count": Project.objects.count(),
             "open_count": Project.objects.filter(status="Open").count(),
             "closed_count": Project.objects.filter(status="Closed").count(),
-        })
\ No newline at end of file
+        })
+    
+    @action(detail=False, methods=["post"])
+    def add_samples_to_study(self, request, pk=None):
+        sample_ids = request.data.get("sample_ids")
+        project_id = request.data.get("project_id")
+        study_letter = request.data.get("study_letter")
+        step_order = request.data.get("step_order", None)
+
+        samples =  make_generator(Sample.objects.filter(id__in=sample_ids))
+        project = Project.objects.get(id=project_id)
+
+        errors = defaultdict(list)
+        for sample in samples:
+            _errors, _ = add_sample_to_study(sample, project, study_letter, step_order)
+            for key, error in _errors.items():
+                if error:
```

Comment:
You set falsy errors in add_sample_to_study?

Reply:
empty list

Reply:
that might be a bug

Reply:
Mmm you may get key = [] sometime when we call error["patate"] = do_that(). it would still extend correctly i think though.

Reply:
Ah now I remember why. It's so I add add any new entries in the `errors` when the error list is empty.